app/workers/scheduler/indexing_scheduler.rb

```diff
@@ -6,17 +6,20 @@ class Scheduler::IndexingScheduler
 
   sidekiq_options retry: 0
 
+  IMPORT_BATCH_SIZE = 100
+  SCAN_BATCH_SIZE = 100 * IMPORT_BATCH_SIZE
+
   def perform
     return unless Chewy.enabled?
 
     indexes.each do |type|
       with_redis do |redis|
-        ids = redis.smembers("chewy:queue:#{type.name}")
-
-        type.import!(ids)
+        redis.sscan_each("chewy:queue:#{type.name}", count: SCAN_BATCH_SIZE).each_slice(IMPORT_BATCH_SIZE) do |ids|
+          type.import!(ids)
 
-        redis.pipelined do |pipeline|
-          ids.each { |id| pipeline.srem("chewy:queue:#{type.name}", id) }
+          redis.pipelined do |pipeline|
+            ids.each { |id| pipeline.srem("chewy:queue:#{type.name}", id) }
+          end
```

Comment:
I wonder if it wouldn't be better to pass all `ids` at once.

I also wonder if it wouldn't be better to rewrite the whole thing to use `spop` instead. On the one hand, a reason not to use it would be error handling, but on the other hand if errors pile up it does not make sense to let the Redis set grow uncontrollably either.

Reply:
I wondered why the original didn't use the batch form of `SREM`. Maybe there's no reason?

The `SPOP` approach could work, and prevents the queue from growing without bound, especially in scenarios where an error is unrecoverable.

The one wrinkle there is that the existing code handles short transient ES outages well, and I wouldn't want to change that property casually if instances are accidentally relying on it.

Reply:
FWIW, Universeodon didn't suffer any obvious ill effects from having 2M status IDs queued for indexing, and once the bug was fixed and this patch was applied, it cleared the backlog successfully.

Just pushed a change to make the import batch size larger and use batched `SREM`, since neither would change the behavior on error.